Update rebus version in order to enable update to net8

MohamedR-Gantner opened the pull request:
In the `Enviso.Pigeon` package there is a dependency on `Rebus.ServiceProvider`.
That [package](https://www.nuget.org/packages/Rebus.ServiceProvider/8.4.0#dependencies-body-tab) has a range dependency of that stops at net8.
Updating that causes another version conflict with the main Rebus package, caused by a version range defined in the Idempotency package.

[List](https://github.com/rebus-org/Rebus/blob/master/CHANGELOG.md#700) of the Rebus changelog, the main breaking change is the default JSON serializer change.

Tests still ran successfully.
Beta packages have been created in order to validate against.

MohamedR-Gantner committed:
Add Azure Artifacts package source, disable MyGet

MohamedR-Gantner committed:
Add Directory.Build.props

MohamedR-Gantner committed:
Add global.json

MohamedR-Gantner committed:
Split in two solution folders

MohamedR-Gantner committed:
Add ruleset to exclude rules

MohamedR-Gantner committed:
Add solution folder for miscellaneous files

MohamedR-Gantner committed:
update ruleset

MohamedR-Gantner committed:
Add docker-compose in order to run the tests using `docker compose run app dotnet test --configuration Release`

MohamedR-Gantner committed:
Remove global.json & analyzers in order to allow the test to run in the container - all packages can be resolved through NuGet

MohamedR-Gantner committed:
Remove duplicate targetFramework definitions

MohamedR-Gantner committed:
Update non-package projects to net8 + add back global.json 🤞

MohamedR-Gantner committed:
Make mysql server configurable based on env var in order to run tests in docker

MohamedR-Gantner committed:
Bump Rebus version up to v8
Changelog, starting from v7: https://github.com/rebus-org/Rebus/blob/master/CHANGELOG.md#700

MohamedR-Gantner committed:
Use shorter alias to set test config

MohamedR-Gantner committed:
Add temp beta suffix

MohamedR-Gantner committed:
Mark packages as packable

arnecornillie-gantner commented on docker-compose.yml:
```diff
@@ -0,0 +1,32 @@
+      - fiber
+
+  mysql:
+    image: mysql:8.0.32
```
Maybe new 8.0.37? Just suggestion :)

MohamedR-Gantner replied:
Done 👌

MohamedR-Gantner committed:
Update Rebus.MySql to the latest version, updating it to use MySqlConnector instead of the MySql.Data
The two packages use pretty much the same API so it should be a drag & drop replacement.

MohamedR-Gantner committed:
Update mysql docker image version to v8.0.37

MohamedR-Gantner committed:
Remove the app portion from the docker-compose, run tests locally
This way it is possible to include back the nuget.config

Commit:
Remove beta version and release v2.3.0
